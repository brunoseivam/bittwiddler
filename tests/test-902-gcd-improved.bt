func gcd(a:uint64, b:uint64):uint64  {
    return match b {
        0 -> { a;             }
        _ -> { gcd(b, a % b); }
    };
}

main {
    var a : uint64;
    var b : uint64;
    var r1 = gcd(a, b);
    var r2 = gcd(b, a);
    emit("gcd({a}, {b}) = {r1}\n");
    emit("gcd({b}, {a}) = {r2}\n");
}
