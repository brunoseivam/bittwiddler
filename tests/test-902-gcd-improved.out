gcd(10, 5) = 5
gcd(5, 10) = 5
